[IMP] new design : left menu reorganization

tivisse opened the pull request:
TASK : https://www.odoo.com/web#id=12927&view_type=form&model=project.task&action=333&active_id=248
PAD : https://pad.odoo.com/p/r.b68kIBheop1ptvzQ

tivisse commented on addons/crm/security/crm_security.xml:
```diff
@@ -42,7 +42,7 @@
         <field name="groups_id" eval="[(4,ref('base.group_sale_manager')),(4,ref('base.group_sale_salesman'))]"/>
     </record>
 
-    <record model="ir.ui.menu" id="base.menu_base_config">
```
Maybe a bad idea to change the xmlid ?

tivisse commented on addons/hr_attendance/hr_attendance_view.xml:
```diff
@@ -107,16 +107,19 @@
+        <menuitem id="timesheet_menu_root" name="Timesheets" sequence="10" icon="fa-user" groups="base.group_user,base.group_hr_user,base.group_hr_manager"/>
+        <menuitem id="menu_hr_time_tracking" name="Time Tracking" parent="timesheet_menu_root" sequence="5" />
 
-        <menuitem id="menu_hr_attendance" name="Attendances" parent="hr.menu_hr_root" sequence="10" groups="base.group_hr_attendance"/>
```
Maybe a bad idea to change the xmlid ?

tivisse commented on addons/hr_evaluation/hr_evaluation_view.xml:
```diff
@@ -71,9 +71,10 @@
             <field name="context">{'search_default_user_to_review_id': [active_id], 'default_user_to_review_id': active_id}</field>
         </record>
 
-        <menuitem name="Appraisal" parent="hr.menu_hr_root" id="menu_eval_hr" sequence="25"/>
```
Maybe a bad idea to change the xmlid ?

tivisse commented on addons/hr_expense/report/hr_expense_report_view.xml:
```diff
@@ -67,7 +67,7 @@
         <field name="search_view_id" ref="view_hr_expense_report_search"/>
     </record>
 
-    <menuitem action="action_hr_expense_report_all" id="menu_hr_expense_report_all" parent="hr.menu_hr_reporting" sequence="1"/>
```
Maybe a bad idea to change the xmlid ?

tivisse commented on addons/hr_payroll/hr_payroll_view.xml:
```diff
@@ -3,9 +3,9 @@
         <!-- Root Menus -->
 
-        <menuitem id="payroll_configure" parent="hr.menu_hr_configuration" name="Payroll" groups="base.group_hr_manager" sequence="45"/>
-        <menuitem id="menu_hr_root_payroll" parent="hr.menu_hr_root" name="Payroll" sequence="30"/>
```
Maybe a bad idea to change the xmlid ?

tivisse commented on addons/hr_timesheet_sheet/hr_timesheet_sheet_view.xml:
```diff
@@ -214,8 +214,9 @@
             </field>
         </record>
 
-        <menuitem action="act_hr_timesheet_sheet_form" id="menu_act_hr_timesheet_sheet_form" parent="hr_attendance.menu_hr_time_tracking"
```
Maybe a bad idea to change the xmlid ?

tivisse commented on addons/marketing_campaign/marketing_campaign_view.xml:
```diff
@@ -184,9 +184,11 @@
         </field>
     </record>
 
-    <menuitem name="Lead Automation" id="menu_marketing_campaign" parent="base.marketing_menu" groups="marketing.group_marketing_user,marketing.group_marketing_manager" 
```
Maybe a bad idea to change the xmlid ?

tivisse commented on addons/mass_mailing/views/mass_mailing.xml:
```diff
@@ -645,11 +646,11 @@
             <field name="view_mode">tree,form</field>
         </record>
 
-        <menuitem name="Mass Mailing" id= "menu_view_mass_mailing_config"
```
Maybe a bad idea to change the xmlid ?

tivisse commented on addons/point_of_sale/point_of_sale_view.xml:
```diff
@@ -989,8 +987,8 @@
 
         <menuitem action="action_pos_session_opening"
-            parent="menu_point_of_sale"
-            id="menu_pos_session_opening" sequence="0"/>
```
Maybe a bad idea to change the xmlid ?

tivisse commented on addons/product_margin/product_margin_view.xml:
```diff
@@ -86,8 +86,8 @@
             </field>
         </record>
 
-  <menuitem id="base.next_id_73" name="Purchase" parent="base.menu_reporting" sequence="36"/>
```
Maybe a bad idea to change the xmlid ?

tivisse commented on addons/project/res_config_view.xml:
```diff
@@ -102,8 +102,8 @@
             <field name="target">inline</field>
         </record>
 
-        <menuitem id="base.menu_project_config" name="Project" parent="base.menu_config"
```
Maybe a bad idea to change the xmlid ?

tivisse commented on addons/purchase/report/purchase_report_view.xml:
```diff
@@ -94,21 +94,9 @@
         <field name="help">Purchase Analysis allows you to easily check and analyse your company purchase history and performance. From this menu you can track your negotiation performance, the delivery performance of your suppliers, etc.</field>
     </record>
 
-    <menuitem id="base.next_id_73" name="Purchase" parent="base.menu_reporting" sequence="37"
```
Maybe a bad idea to change the xmlid ?

tivisse commented on addons/purchase/res_config_view.xml:
```diff
@@ -104,8 +104,8 @@
             <field name="target">inline</field>
         </record>
 
-        <menuitem id="menu_purchase_config" name="Purchases" parent="base.menu_config"
```
Maybe a bad idea to change the xmlid ?

tivisse commented on addons/sales_team/sales_team.xml:
```diff
@@ -191,7 +191,7 @@
             </xpath>
         </template>
 
-        <menuitem id="base.next_id_64" name="Sales" parent="base.menu_reporting" sequence="1" groups="base.group_sale_manager"/>
```
Maybe a bad idea to change the xmlid ?

tivisse commented on addons/stock/stock_view.xml:
```diff
@@ -1774,13 +1776,13 @@
-        <menuitem id="menu_stock_procurement" name="Automatic Procurements" parent="stock.menu_stock_configuration" sequence="5"/>
+        <menuitem id="menu_stock_procurement" name="Automatic Procurements" parent="stock.menu_stock_config_settings" sequence="5"/>
         <menuitem action="action_orderpoint_form" id="menu_stock_order_points" parent="stock.menu_reordering_rules_config" sequence="1"/>
-        <menuitem id="stock.next_id_61" name="Warehouse" sequence="40" parent="base.menu_reporting" groups="group_stock_manager"/>
```
Maybe a bad idea to change the xmlid ?

rim-odoo commented:
@tivisse  Yes, avoid changing xmlids, as it will add migration work

rim-odoo commented:
@tivisse Can you make a more complete commit message listing the main modules/files split/moves ?

tivisse commented:
@rim-odoo, @tde-banana-odoo seems ok to change/update/give_a_super_cute xmlid :)
Generally, I created new xmlid when I splitted some modules into several apps. 
I think that I have changed some ridiculous xmlid as next_id_64 or something like this. 

By the way, I have updated the commit message, I hope it is good enough ;)

rim-odoo commented on addons/website_slides/__openerp__.py:
```diff
@@ -31,4 +31,6 @@
         'data/website_slides_demo.xml'
     ],
     'installable': True,
+    'auto_install': False,
```
default value...

rim-odoo commented on addons/website_links/static/src/js/website_links_backend.js:
```diff
@@ -0,0 +1,21 @@
+odoo.define('web.website_links_embedded', function (require) {
+"use strict";
+var core = require('web.core');
+var Widget = require('web.IFrameWidget');
```
Perfect example of bad design: add an iFrame of the frontend inside the backend...

rim-odoo commented on addons/web/views/webclient_templates.xml:
```diff
@@ -196,8 +196,9 @@
-                                <t t-esc="menu['name']"/>
+                            <div class="oe_secondary_menu_section">                                
+                                <t t-if="menu['children']"><t t-esc="menu['name']"/></t>
+                                <t t-if="not menu['children']"><t t-call="web.menu_link"/></t>
```
Clickable secondary menus are ugly... need a restyling?

tivisse replied:
Will be done in the new design task.

rim-odoo commented on addons/project_issue/project_issue.py:
```diff
@@ -34,17 +34,6 @@
-    _defaults = {
-        'active': 1,
-    }
-
```
:+1:

rim-odoo commented on addons/mass_mailing/views/res_partner.xml:
```diff
@@ -1,6 +0,0 @@
-    <data>
-
-    </data>
-</openerp>
```
:+1:

rim-odoo commented on addons/contacts/__openerp__.py:
```diff
@@ -0,0 +1,44 @@
+    'auto_install': False,
+}
+
+# vim:expandtab:smartindent:tabstop=4:softtabstop=4:shiftwidth=4:
```
modelines

rim-odoo commented on addons/contacts/__openerp__.py:
```diff
@@ -0,0 +1,44 @@
+##############################################################################
+#
+#    OpenERP, Open Source Management Solution
+#    Copyright (C) 2004-2010 Tiny SPRL (<http://tiny.be>).
```
Odoo ?

rim-odoo commented on addons/contacts/__openerp__.py:
```diff
@@ -0,0 +1,44 @@
+    ],
+    'installable': True,
+    'application': True,
+    'auto_install': False,
```
auto_install: False is the default value

rim-odoo commented on addons/contacts/__openerp__.py:
```diff
@@ -19,4 +19,23 @@
+This module gives you a quick view of your address book, accessible from your home page.
+You can track your suppliers, customers and other contacts.
+""",
+    'author': 'OpenERP SA',
```
Odoo SA ? ;-)
